tt_eager/tt_numpy/functions.hpp

```diff
@@ -368,6 +368,45 @@ static Tensor index_all(
     return output;
 }
 
+template <typename T>
+static Tensor mask_padded_input(
+    const Shape& padded_shape,
+    const Shape& unpadded_shape,
+    DataType data_type,
+    const Layout layout = Layout::ROW_MAJOR,
+    Device* device = nullptr,
+    const MemoryConfig& output_mem_config = MemoryConfig{
+        .memory_layout = tt::tt_metal::TensorMemoryLayout::INTERLEAVED}) {
+    auto owned_buffer = tt_metal::owned_buffer::create<T>(tt_metal::compute_volume(padded_shape));
```

Comment:
If you use the `tt_metal::owned_buffer::create<T>(std::vector<T>&& storage)` constructor, then you can pass in your own vector initialized with the specific mask value you want, and you can also update the for loops to iterate on the unpadded shapes since you no longer need the if condition.